cpp/src/arrow/compute/exec/sink_node.cc

```diff
@@ -175,9 +175,9 @@ struct OrderBySinkNode final : public SinkNode {
                                            plan()->exec_context()->memory_pool());
     if (ErrorIfNotOk(maybe_batch.status())) {
       StopProducing();
-      bool cancelled = input_counter_.Cancel();
-      DCHECK(cancelled);
-      finished_.MarkFinished(maybe_batch.status());
+      if (input_counter_.Cancel()) {
```

Comment:
Hmm, is it possible for input_counter_.Cancel() to be false?

Reply:
Looks like is possible. Some runs in stress mode (`stress -m 16`) enable `DCHECK(cancelled);`. So a checks is needed.